inc/class_parser.php

```diff
@@ -1605,18 +1605,27 @@ function mycode_parse_video_disabled_callback($matches)
 	*/
 	function mycode_auto_url($message)
 	{
-		$message = " ".$message;
-
-		if($this->options['allow_url'] != 1)
-		{
-			return $message;
-		}
-		
 		// Links should end with slashes, numbers, characters and braces but not with dots, commas or question marks
 		// Don't create links within existing links (handled up-front in the callback function).
-		$message = preg_replace_callback("#<a\\s[^>]*>.*?</a>|([\s\(\)\[\>])(http|https|ftp|news|irc|ircs|irc6){1}(://)([^\/\"\s\<\[\.]+\.([^\/\"\s\<\[\.]+\.)*[\w]+(:[0-9]+)?(/([^\"\s<\[]|\[\])*)?([\w\/\)]))(?![^<>]*?>)#ius", array($this, 'mycode_auto_url_callback'), $message);
-		$message = preg_replace_callback("#<a\\s[^>]*>.*?</a>|([\s\(\)\[\>])(www|ftp)(\.)(([^\/\"\s\<\[\.]+\.)*[\w]+(:[0-9]+)?(/([^\"\s<\[]|\[\])*)?([\w\/\)]))(?![^<>]*?>)#ius", array($this, 'mycode_auto_url_callback'), $message);
-		$message = my_substr($message, 1);
+		$message = preg_replace_callback(
+			"~
+				<a\\s[^>]*>.*?</a>|								# match and return existing links
+				(?<=^|[\s\(\)\[\>])								# character preceding the link
+				(?P<prefix>
+					(?:http|https|ftp|news|irc|ircs|irc6)://|	# scheme, or
+					(?:www|ftp)\.								# common subdomain
+				)
+				(?P<link>
+					[^\/\"\s\<\[\.]+\.*[\w]+					# host
```

Comment:
Parentheses are missing here around the pattern prior to the wildcard character, which should apply to the whole preceding match, not just the literal dot. The effect of this is that only two-part domains are currently matched.